log/scope_test.go

```diff
@@ -218,6 +220,89 @@ func TestBasicScopes(t *testing.T) {
 	}
 }
 
+func TestScopeWithLabel(t *testing.T) {
+	const name = "TestScope"
+	const desc = "Desc"
+	s := RegisterScope(name, desc, 0)
+	s.SetOutputLevel(DebugLevel)
+
+	lines, err := captureStdout(func() {
+		funcs.Store(funcs.Load().(patchTable))
+		s2 := s.WithLabels("foo", "bar").WithLabels("baz", 123, "qux", 0.123)
+		s2.Debuga("Hello")
+		// s should be unmodified.
+		s.Debuga("Hello")
+
+		_ = Sync()
+	})
+	if err != nil {
+		t.Errorf("Got error '%v', expected success", err)
+	}
+
+	mustRegexMatchString(t, lines[0], `Hello {"foo": "bar","baz": 123, "qux": 0.123}`)
+	mustRegexMatchString(t, lines[1], `Hello$`)
+}
+
+func TestScopeWithoutLabels(t *testing.T) {
+	const name = "TestScope"
+	const desc = "Desc"
+	s := RegisterScope(name, desc, 0)
+	s.SetOutputLevel(DebugLevel)
+
+	lines, err := captureStdout(func() {
+		funcs.Store(funcs.Load().(patchTable))
+		s = s.WithLabels("foo", "bar", "baz", 123)
+		s2 := s.WithoutLabels("baz")
+		s.Debuga("Hello")
+		s2.Debuga("Hello")
+		s2 = s.WithoutAnyLabels()
+		s2.Debuga("Hello")
+
+		_ = Sync()
+	})
+	if err != nil {
+		t.Errorf("Got error '%v', expected success", err)
+	}
+
+	mustRegexMatchString(t, lines[0], `Hello {"foo":"bar","baz":123}`)
+	mustRegexMatchString(t, lines[1], `Hello {"foo":"bar"}`)
+	mustRegexMatchString(t, lines[2], `Hello$`)
+}
+
+func TestScopeErrorDictionary(t *testing.T) {
+	const name = "TestScope"
+	const desc = "Desc"
+	s := RegisterScope(name, desc, 0)
+	s.SetOutputLevel(DebugLevel)
+
+	ie := &structured.Error{
+		MoreInfo:    "MoreInfo",
+		Impact:      "Impact",
+		Action:      "Action",
+		LikelyCause: "LikelyCause",
+	}
+	lines, err := captureStdout(func() {
+		funcs.Store(funcs.Load().(patchTable))
+		s.WithLabels("foo", "bar").Debuga(ie, "Hello")
+
+		_ = Sync()
+	})
+	if err != nil {
+		t.Errorf("Got error '%v', expected success", err)
+	}
+
+	mustRegexMatchString(t, lines[0], `Hello	{"message": "Hello", "moreInfo": "MoreInfo", "impact": "Impact", "action": "Action", "likelyCauses": "LikelyCause", "foo": "bar"}`)
```

Comment:
Where does the big space between Hello and `{` come from? Or is a tab and it displays weird on the diff?

Reply:
these are both due to switching to the zap json encoder. it's their fault now!